Python/msrd.py

```diff
@@ -29,48 +37,149 @@ def __init__(self, msrd_url, account_id, api_token):
+                        r.status_code,
+                        r.reason,
+                        r.url
+                    )
+                )
+            }
+        )
 
     def _url(self, fmt, *args, **kwargs):
+        """
+        _url: Format URL.
+        """
         path = fmt.format(*args, **kwargs)
         return urljoin(self.msrd_url, path)
 
     def account_info(self):
+        """
+        Get Account Information via a REST Call to the MSRD API.
+        """
         url = self._url('api/accounts/{}', self.account_id)
         return self.session.get(url)
 
     def os_images(self):
+        """
+        Get Available OS Images for the Account via a REST Call to the MSRD API.
+        """
         url = self._url('api/accounts/{}/osimages', self.account_id)
         return self.session.get(url)
 
     def job_tiers(self):
+        """
+        Get Available Job Tiers for the Account via a REST Call to the MSRD API.
+        """
         url = self._url('api/accounts/{}/jobtiers', self.account_id)
         return self.session.get(url)
 
     def jobs(self):
+        """
+        Get Available Job for the Account via a REST Call to the MSRD API.
+        """
         url = self._url('api/accounts/{}/jobs', self.account_id)
         return self.session.get(url)
 
     def upload_file(self, file_path):
+        """
+        Upload file via a REST Call to the MSRD File PUT API.
+        """
         url = self._url('files/accounts/{}/session', self.account_id)
-        with open(file_path, 'rb') as f:
-            return self.session.put(url, data=f)
+        with open(file_path, 'rb') as file_to_upload:
+            return self.session.put(url, data=file_to_upload)
 
     def submit_job(self, job):
+        """
+        Submit Job via a REST Call to the MSRD API.
+        """
         url = self._url('api/accounts/{}/jobs', self.account_id)
         return self.session.post(url, json=job)
 
 
-def print_response(r):
+def print_response(response):
+    """
+    Pretty-Print response data
+    """
     try:
-        print(json.dumps(r.json(), indent=2))
+        print(json.dumps(response.json(), indent=2))
     except json.JSONDecodeError:
-        print(r.text)
+        print(response.text)
+
+
+DEFAULT_MSRD_URL = 'https://microsoftsecurityriskdetection.com'
 
+MAX_FILE_SIZE = 4194304  # 4 Megabytes.
+
+
+def generate_file_info(client, file):
+    """
+    Generate file information required by the job submission:
+        URL: The URL generated by the upload of the resource.
+    """
+    path = Path(file)
+    size = path.stat().st_size
+
+    if size > MAX_FILE_SIZE:
+        print('ERROR: file "{}" has byte size {}, which exceeds limit of 4mb'.format(file, size))
+        exit(1)
+
+    name = path.name
+
+    # Does not currently return JSON, but a double-quoted URL as text.
+    # Remove the double-quotes to avoid errors later on.
+    url = client.upload_file(file).text.strip('"')
+    return name, url
```

Comment:
Let's have this function return a dict:
```python
        {
            'action': 'DownloadOnly',
            'name': name,
            'url': url,
        }
```
The `fileInformations` field has that awkward forced-plural name because each value in it is a `fileInformation` object.

Reply:
Done